x-pack/plugins/maps/public/routing/page_elements/top_nav_menu/top_nav_menu.js

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  getNavigation,
+  getCoreChrome,
+  getMapsCapabilities,
+  getInspector,
+  getToasts,
+  getCoreI18n,
+  getData,
+  getUiSettings,
+} from '../../../kibana_services';
+import {
+  SavedObjectSaveModal,
+  showSaveModal,
+} from '../../../../../../../src/plugins/saved_objects/public';
+import { MAP_SAVED_OBJECT_TYPE } from '../../../../common/constants';
+import { updateBreadcrumbs } from '../breadcrumbs';
+import { goToSpecifiedPath } from '../../maps_router';
+
+/**
+ * @return {null}
+ */
+export function MapsTopNavMenu({
+  savedMap,
+  query,
+  onQueryChange,
+  onQuerySaved,
+  onSavedQueryUpdated,
+  savedQuery,
+  time,
+  refreshConfig,
+  setRefreshConfig,
+  setRefreshStoreConfig,
+  initialLayerListConfig,
+  isVisible,
+  indexPatterns,
+  updateFiltersAndDispatch,
+  isSaveDisabled,
+  closeFlyout,
+  enableFullScreen,
+  openMapSettings,
+  syncSavedMap,
+  inspectorAdapters,
+  syncAppAndGlobalState,
+  currentPath,
+}) {
+  const { TopNavMenu } = getNavigation().ui;
+  const { filterManager } = getData().query;
+  const showSaveQuery = getMapsCapabilities().saveQuery;
+  const onClearSavedQuery = () => {
+    onQuerySaved(undefined);
+    onQueryChange({
+      filters: filterManager.getGlobalFilters(),
+      query: {
+        query: '',
+        language: getUiSettings().get('search:queryLanguage'),
+      },
+    });
+  };
+
+  // Nav settings
+  const [isOpenSettingsDisabled, setIsOpenSettingsDisabled] = useState(false);
```

Comment:
Nothing is ever calling `setIsOpenSettingsDisabled ` so map settings button is not getting disabled when flyout is open.

<img width="600" alt="Screen Shot 2020-06-15 at 8 17 41 AM" src="https://user-images.githubusercontent.com/373691/84668340-c341d400-aee0-11ea-8116-2726836aa5b6.png">